### ISSUE TITLE
Question about OWL to OBO rdfs:label mappings

### ISSUE THREAD

**Issue by fanavarro** [owlcs/owlapi#861]
Hi all,

Currently I am testing some conversions from owl to obo format. In this case I was testing [plant ontology](https://www.ebi.ac.uk/ols/ontologies/po) and it finish with an error because some classes are annotated with several rdfs:label. In the conversion to obo, rdfs:label are converted into name tag. If you have more than one label, the resulting obo term would have more than one name, which is not allowed. After the conversion is done, this kind of restrictions are checked, throwing an exception in this case due to multiple name tags for the same term.

I am wondering if it could make sense to translate extra owl labels into another obo tag instead of throwing an exception. It would be perfect if something like ALT_NAME exists, but it does not. I think the best alternative would be a synonym with a specific type.

What do you think about this approach? Does it have any sense for you? I could work on it, but I wanted to have your feedback before.

Regards and thanks for your time.

**Comment by matentzn** [comment]
I agree with Francisco that this should be dealt with; I think actually, double name tags should not be serialised at all. We have [discussed this issue](https://github.com/INCATools/ontology-development-kit/issues/230) previously in the context of the ODK; I believe the way to go here is to push all labels beyond the first into the `owl-axioms:` section of the OBO file (alphanumeric sort order on label string.

While the suggestion here is surely interesting from a user-perspective, it would violate any pretence of roundtripability :P Thanks for raising this!

**Comment by fanavarro** [comment]
Hi @matentzn! Thanks for linking my issue to your discussion.

In my opinion, labels are important in owl ontologies, and moving part of them to untranslatable owl axioms could mean a loss of relevant information in the resulting obo file.

I know that my solution is a little bit "hacky" as I am creating a new synonym type definition that could coincide with someone of those defined by the user. Nonetheless, this is the best solution I have found for having this information in the final file. I also think that it would be so difficult to change obo spec to add an ALT_NAME tag or to modify obo in owl mappings to add this special synonym type for storing extra labels. I am also curious about @althonos suggestion of merging the labels into one statement. How would you do that?

I must admit that your solution is the best one for a theoretical point of view. Finally. I think we agree in that this case should be treated and the translation from owl to obo should not finish in an exception.

Greetings and thanks again!

**Comment by althonos** [comment]
Actually @fanavarro your suggestion makes more sense, it's probably best to use a built-in `EXACT` synonym type which would translate back as a `rdfs:label` to allow roundtripping. (or maybe use qualifiers as these were made for that).

**Comment by matentzn** [comment]
I must admit I am not too invested in OBO; so I am happy if defining a new property makes you happy. As long as it does not interact at all with the official "has_exact_synynonym" property, it would be roundtripable, which is the only thing I care about (in particular, any obo serialised should be legal obo). Who would be the right person to write such an extension?
